### ISSUE TITLE
If you block federation with bookwyrm.social, book covers aren't imported for new books

### ISSUE THREAD

**Issue by IcedEagle** [bookwyrm-social/bookwyrm#1379]
**Describe the bug**
If you decide to block the bookwyrm.social server from being federated (in my case, I'm on my own island), book covers will never get imported for books even when there are book covers on OpenLibrary and Inventaire for a book.

**To Reproduce**
1. Block federation with bookwyrm.social
2. Import a book that has a cover on OpenLibrary and Inventaire
3. Notice the book on your instance does not have a cover

**Expected behavior**
If the book cover exists on other sources, they are used for importing as well.

**Instance**
My personal instance.

**Additional context**
I made sure to double check that the bookwyrm.social connector was disabled, and I also set its priority to be lower than OpenLibrary and Inventaire but that did not seem to help.

(Apologies as I feel a bit bad filing bugs like this since I'm sure preference is to federate! Hopefully it doesn't come off the wrong way...)

---

**Desktop (please complete the following information):**
    - OS: Windows 11
    - Browser Firefox
    - Version 94

**Comment by mouse-reeve** [comment]
Not federating is totally fine, zero offense taken - I am curious why you don't want to though, if you're interested in getting into that.

It's possible that the connector isn't being properly disabled somewhere. When you say import, can you give me more details about what the steps are? Is this a csv data import, or are you referring to importing from the search page?

If you're coming from the search page, are you seeing search results from bookwyrm.social, and which connector are you importing from?

**Comment by IcedEagle** [comment]
It's specifically from the search page, and I'm importing from OpenLibrary. On the search page I don't see bookwyrm.social results.

Screenshots of what I'm seeing (notice it has a cover, and on the Open Library page I can also see the cover)):
![image](https://user-images.githubusercontent.com/934801/132636784-3f6f5877-0f48-4f4e-a099-9bddca3fb474.png)

Click "Import Book"

Result:
![image](https://user-images.githubusercontent.com/934801/132636862-7c3f13f6-fcc5-4946-864c-f5bb790b6980.png)

In the Django admin page, this is what the Connector shows after blocking federation:
![image](https://user-images.githubusercontent.com/934801/132637176-93642cb5-40e0-4806-a138-74ddc03ae84b.png)

**Comment by IcedEagle** [comment]
The "why not federate" is easy for me to answer. Bookwyrm is the first federated software I've stood up at home, and to start I'm interested in keeping a private social bubble and making this more like a private Goodreads for my partner and I, rather than a step into a larger social network. Right now my instance is for myself, my partner and I'll likely invite a few friends onto the instance at some point, maybe 5 people tops. It's pretty nice that right now all of the posts are just me and my partner, so it's very minimalist. I don't have Mastodon or other federated social sites yet, but I am interested in checking them out at some point!

We looked at a lot of various Goodreads alternatives, stumbled on Bookwyrm and I thought it was easily the best looking alternative, and very much heading in the direction of what we'd want. I think one of the cool parts of the federated systems is the flexibility they offer as an admin. Right now I'm on my own island, but if I want to open it up into the broader next and be a bit more extraverted it's pretty much no work to do that! Or I can pick and choose which instances I would want to federate with, which I think is really cool.

I could totally be doing this federated server thing wrong, and maybe there's better ways for me to handle this. For example, maybe if we default our posts to Followers only, don't follow anyone else, and don't make our profiles discoverable we can feel like we're in an island, while still being federated. At that point, I figured why not just block federation if neither me or my partner intend to go be a bit more social... At least it's very easy to change.

Hopefully that provides insight into my use case. :) Feel free to poke at any of that if you want, I'm very new to federated services. Appreciate the support as always!

**Comment by mouse-reeve** [comment]
I think this is unrelated to federation. You'll get the same results when you import from OpenLibrary regardless of your federation status with b.s -- the problem is the OpenLibrary serves preview images that aren't associated with the data that gets imported. OpenLibrary stores a cover image associated with the "work", and that is the image that's presented in search. It may be the case that none of the editions of that work have a cover set (like with _The House in the Cerulean Sea_), so even though the search results look like you'll get a cover, you don't.

I think the best solution is to use the work cover for the default edition that's imported when that edition doesn't have a cover of its own?

And thanks for the explanation -- #559 will be one for you I think :)

**Comment by IcedEagle** [comment]
Interesting, thanks for taking a look! I think what you described makes sense so during import its preferred to have a cover versus not having one. At least, from my naive viewpoint, I saw there was a cover on OpenLibrary, and one didn't get imported. I ran a test with and without federation, but I used two different books that I thought looked identical in setup, but that very likely wasn't the case... I turned federation back on my server the other day, did an import of new books today and sure enough, many of them still didn't have covers so yeah, federation isn't directly related.

From what I can tell, the specific edition that was imported for _The House in the Cerulean Sea_ does have a cover image on OpenLibrary? But maybe I'm not understanding the OpenLibrary website with editions vs works and how it shows their covers or something like that.

This is the specific edition that was imported for reference: https://openlibrary.org/books/OL28281265M/The_House_in_the_Cerulean_Sea
